src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2789,7 +2796,23 @@ FindAllowedTargetFillState(RebalanceState *state, uint64 shardId)
 				targetFillState->node,
 				state->functions->context))
 		{
-			return targetFillState;
+			if (ShardExists(shardId))
```

Comment:
It doesn't look nice to me as well, but added it for the tests we have in `shard_rebalancer_unit.sql`. As the shards in those tests do not actually exist, they all fail without this check.

Reply:
added it because `ActiveShardPlacementListOnGroup` errors out for non-existing shards. After applied your suggestion in the other comment, we don't need it. Removed.